cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -164,6 +178,18 @@ func (s *Service) deleteInternalLoadBalancer(ctx context.Context, name string) e
 	}
 	s.scope.Network().APIInternalHealthCheck = nil
 
+	lbSpec := s.scope.LoadBalancer()
+	createBootstrapInstanceGroup := false
+	if lbSpec.InternalLoadBalancer != nil {
```

Comment:
Do we need the if statement here? With the default value set to false on deref it may be redundant.

Reply:
Good point. Removed statements.